core/blockchain.go

```diff
@@ -2110,6 +2155,9 @@ func (bc *BlockChain) insertChain(chain types.Blocks, verifySeals bool) (int, er
 		if err != nil {
 			return it.index, err
 		}
+		if statedb.NoTrie() {
+			statedb.SetCurrentRoot(block.Root())
```

Comment:
Please reuse the `expectedRoot` in  `StateDB`

Reply:
> Please reuse the `expectedRoot` in `StateDB`

fixed